libs/content/workspace/src/workspace.rs

```diff
@@ -540,24 +540,16 @@ impl Workspace {
                     let core = self.core.clone();
                     let update_tx = self.updates_tx.clone();
                     let ctx = self.ctx.clone();
-                    let seq = if let Some(TabContent::Markdown(md)) = &tab.content {
-                        md.editor.buffer.current.seq
-                    } else {
-                        0
-                    };
 
                     thread::spawn(move || {
-                        let content = save_req.content;
-                        let id = save_req.id;
-                        let hmac = save_req.hmac;
+                        let SaveRequest { seq, content, id, old_hmac, safe_write } = save_req;
 
-                        let result = if hmac.is_some() {
-                            core.safe_write(id, hmac, content.clone().into())
-                                .map(|hmac| (content, Some(hmac), Instant::now(), seq))
+                        let result = if safe_write {
```

Comment:
On master, when the initial content was empty, the document would be written unsafely. This is why the issue is only reproducible for new files before they have been saved. This issue is fixed using the explicit safe_write flag.